src/bin/scryer-prolog.rs

```diff
@@ -23,6 +23,6 @@ fn main() -> std::process::ExitCode {
 
     runtime.block_on(async move {
         let mut wam = machine::Machine::new(Default::default());
-        wam.run_top_level(atom!("$toplevel"), (atom!("$repl"), 1))
+        wam.run_module_predicate(atom!("$toplevel"), (atom!("$repl"), 0))
```

Comment:
This is a great change, and paves the way to **flexibly specifying** the goal that should run as toplevel, for example with `-t Goal` on the command line. We could then run scripts with:

<pre>
$ scryer-prolog -t halt -g Goal
</pre>

and ensure that Scryer Prolog *halts* after goal succeeds **or fails or raises an exception**!